### PR TITLE
Improvement api added database schema fixes #18

### LINKED ISSUE #18: Database schema
We need to get a database schema done before we implement the back end any further.
Thinking of setting up the schema in the Wiki or maybe just tables in there.

So looking to generate some discussion for the main purpose for the program and the tables associated.

The following is a quick example

# Tables
* Users

## Users
* Id (PK)
* Username
* Password

### PR DESCRIPTION
**Pull request by MrCeleryman**
I am implemented the database schema as discussed in #18 
Had to make the deleted properties into their own NullTime struct to deal with nullable time

Also changed the DELETE requests on Review, Venues and Users tables to PATCH to comply with REST standards

### CODE REVIEW

File: api/venues.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+	"github.com/gin-gonic/gin"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type Venues struct {
+	ID        int       `gorm:"AUTO_INCREMENT" form:"id" json:"id"`
+	VenueName string    `gorm:"not null;size:255" form:"venueName" json:"venueName"`
+	Address   string    `gorm:"not null" form:"address" json:"address"`
+	Latitude  float64   `gorm:"not null" form:"firstName" json:"latitude"`
```
Comment (JonathanHolland) [inline api/venues.go]: firstName - latitude? Am I missing something here or is this just a typo?

File: api/venues.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+		result := Venues{
+			ID:      venue.ID,
+			Deleted: NullTime{Time: time.Now(), Valid: true},
+		}
+
+		db.Save(&result)
+		c.JSON(201, gin.H{"success": result})
+	} else {
+		c.JSON(404, gin.H{"error": "Venue #" + id + " not found"})
```
Comment (JonathanHolland) [inline api/venues.go:118]: A lot of good work in here Dan, but can I suggest as part of another feature branch adding a static file with all of our error codes, and associated default messages? i.e. ERR_NOT_FOUND, 404, "The requested resource could not be found" (or even with functions built on top of these static descriptors to insert the id etc. variable information into the generic message).

**Review by JonathanHolland — CHANGES_REQUESTED** [review]

**Comment by CaptainAchilles** [comment]
I would like to point out that I also noticed issues with the gorm stuff for the venues table regarding latitude and longitude

File: api/venues.go (hunk 1 above)
Comment (MrCeleryman) [reply api/venues.go]: Good catch I'll fix it up now

File: api/venues.go (hunk 2 above)
Comment (MrCeleryman) [reply api/venues.go:118]: Sounds like an excellent idea, feel free to open an issue or start on this yourself

**Review by JonathanHolland — APPROVED** [review]

**Pull request merged**